YueYingLee/VeriFace
# Attendance

1) ROUTES.PY:
Register function:
- Edited register() function to encode the images that users upload.
- The encoding has several checks such as ensuring correct file extension type, a face can be found in the image, and only one face allowed in the image. If any fails, then image will not be approved.

Start/Stop attendance functions:
- Added start_attendance() and stop_attendance() to start and end attendance threads
- Has 2 threads: one for polling for RFID scans, and the other for keeping the camera open
- The attendance process will start for the chosen event ID only, and will only accept RFID scans from users registered for that event 

2) RFID_HANDLER.PY:
- Constantly poll for RFID scan, then stop scanning once it detects an RFID
- Verify that the RFID is registered for the current event. If so, start facial recognition function
- If facial recognition finishes successfully, mark the attendance, then continue polling for scans again

3) EVENT_HANDLER.PY:
- Just defines the events to control threads

4) RECOGNITION_HANDLER.PY:
- main facial recognition code
- based on the RFID scanned, look for that user's face
- if found, exit facial recognition and return the user for marking attendance
- if nothing found for 10 seconds, timeout and exit to prevent infinite loop

5) RECOGNITION_UTILS.PY:
- helper functions !
- mark_attendance() is still only file based and not saved in database, will change later on

To test this, we need:
- finish setting up the events page and event controls for admins to start/stop the attendance process
- add rfid field for User
- add 'events' field for User model (this should be a list of all event IDs that the user is registered for)
- add 'users' field for Event model (this should be a list of all user IDs that is registered for the specific event)
